warehouse/query-core/src/main/java/datawave/query/transformer/KeywordQueryTransformer.java

```diff
@@ -0,0 +1,123 @@
+import datawave.query.table.parser.KeywordKeyValueFactory;
+import datawave.query.util.keyword.KeywordResults;
+import datawave.webservice.query.result.event.EventBase;
+import datawave.webservice.query.result.event.FieldBase;
+import datawave.webservice.query.result.event.Metadata;
+import datawave.webservice.query.result.event.ResponseObjectFactory;
+import datawave.webservice.result.BaseQueryResponse;
+import datawave.webservice.result.EventQueryResponseBase;
+
+@SuppressWarnings("rawtypes")
+public class KeywordQueryTransformer extends BaseQueryLogicTransformer<Entry<Key,Value>,EventBase> {
+
+    private static final Logger log = Logger.getLogger(KeywordQueryTransformer.class);
+
+    protected final Authorizations auths;
+    protected final ResponseObjectFactory responseObjectFactory;
+    protected final Map<Metadata,String> metadataIdMap;
+    protected final boolean decodeView;
+
+    public KeywordQueryTransformer(Query query, MarkingFunctions markingFunctions, ResponseObjectFactory responseObjectFactory) {
+        this(query, markingFunctions, responseObjectFactory, false);
+    }
+
+    public KeywordQueryTransformer(Query query, MarkingFunctions markingFunctions, ResponseObjectFactory responseObjectFactory, boolean decodeView) {
+        super(markingFunctions);
+        this.auths = new Authorizations(query.getQueryAuthorizations().split(","));
+        this.responseObjectFactory = responseObjectFactory;
+        // the metadata id map allows us to recover document identifiers from shard/datatype/uid as long as they are provided in the query.
+        this.metadataIdMap = ContentQueryTransformer.extractMetadadaIdMap(query);
+        this.decodeView = decodeView;
+    }
+
+    @Override
+    public EventBase transform(Entry<Key,Value> entry) {
+
+        if (entry.getKey() == null && entry.getValue() == null)
+            return null;
+
+        if (null == entry.getKey() || null == entry.getValue()) {
+            throw new IllegalArgumentException("Null key or value. Key:" + entry.getKey() + ", Value: " + entry.getValue());
+        }
+
+        KeywordKeyValueFactory.KeywordKeyValue kkv;
+        try {
+            kkv = KeywordKeyValueFactory.parse(entry.getKey(), entry.getValue(), auths, markingFunctions);
+        } catch (Exception e1) {
+            throw new IllegalArgumentException("Unable to parse visibility", e1);
+        }
+
+        EventBase e = responseObjectFactory.getEvent();
+        FieldBase field = responseObjectFactory.getField();
+
+        e.setMarkings(kkv.getMarkings());
+
+        // capture the metadata that identifies the field.
+        Metadata m = new Metadata();
+        m.setRow(kkv.getShardId());
+        m.setDataType(kkv.getDatatype());
+        m.setInternalId(kkv.getUid());
+        e.setMetadata(m);
+
+        // store the content in a field based on its view name.
+        field.setMarkings(kkv.getMarkings());
+        field.setName(kkv.getViewName() + "_KEYWORDS");
```

Comment:
consider making this configurable

Reply:
Will look into this in the next round of changes